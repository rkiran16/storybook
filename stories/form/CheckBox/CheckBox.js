import { html } from "lit-html";

export const CheckBox = ({checked, isDisabled, isGroup, showLabel = true}) => {
    const itemTemplates = [];
    const items = [1, 2, 3, 4, 5];
    for (const i of items) {
        itemTemplates.push(html`
            <div class="form-check">
                <input type="checkbox" class="form-check-input" ?checked=${checked}  ?disabled=${isDisabled} id="customCheck${i}" />
                <label class=${`form-check-label ${showLabel ? '' : 'visually-hidden'}`} for="customCheck111-33">Check this custom checkbox</label>
            </div>`);
    }
    if (!isGroup) {
        return html`
            <div class="form-check">
                <input type="checkbox" class="form-check-input" ?checked=${checked}  ?disabled=${isDisabled} id="customCheck111-33" />
                <label class=${`form-check-label ${showLabel ? '' : 'visually-hidden'}`} for="customCheck111-33">Check this custom checkbox</label>
            </div>
        `;
    } else {
        return html`
        <fieldset>
            <legend>Group Label</legend>
            ${itemTemplates}
        </fieldset>
        `;
    }
    
}