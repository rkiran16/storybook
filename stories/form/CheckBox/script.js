export default function(args) {
  if (args.indeterminate) {
    return `
      document.getElementById('customCheck111-33').indeterminate = true;
    `;
  }
};