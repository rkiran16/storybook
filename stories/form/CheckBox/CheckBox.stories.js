import { html } from 'lit-html';
import { CheckBox } from './CheckBox';
import script from './script';

export default {
   title: "Forms/CheckBox",
   parameters: {
      zeplinLink: 'https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62a8d982f6be772c3587fe99'
   },
   argTypes: {
      isDisabled: {
         name: "isDisabled",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: { type: 'boolean'}
      },
      checked: {
         name: "checked",
         type: { name: 'boolean', required: false},
         description: "Use when the user cannot proceed or until an input is collected",
         defaultValue: false,
         table: {
            type: { summary: 'boolean' },
            defaultValue: { summary: false },
         },
         control: { type: 'boolean'}
      },
      indeterminate: {
         control: false
      }
   },
   decorators: [(story, context) => {
      return (
         html`${CheckBox(context.args)}
         <script>${script(context.args)}</script>
         `
      );
   }]
}

const CheckBoxTemplate = (args) => CheckBox(args);

export const Standard = CheckBoxTemplate.bind({});

export const Disabled = CheckBoxTemplate.bind({});
Disabled.args = {
   isDisabled: true
}

export const DisabledSelected = CheckBoxTemplate.bind({});
DisabledSelected.args = {
   isDisabled: true,
   checked: true
}

export const Indeterminate = CheckBoxTemplate.bind({});
Indeterminate.args = {
   indeterminate: true
}

export const Group = CheckBoxTemplate.bind({});
Group.args = {
   isGroup: true
}