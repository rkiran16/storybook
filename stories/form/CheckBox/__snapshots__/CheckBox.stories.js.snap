// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Forms/CheckBox Disabled 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "
         <script>",
    "</script>
         ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
        "  ?disabled=",
        " id=\\"customCheck111-33\\" />
                <label class=",
        " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>
        ",
      ],
      "values": Array [
        false,
        true,
        "form-check-label ",
      ],
    },
    undefined,
  ],
}
`;

exports[`Storyshots Forms/CheckBox Disabled Selected 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "
         <script>",
    "</script>
         ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
        "  ?disabled=",
        " id=\\"customCheck111-33\\" />
                <label class=",
        " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>
        ",
      ],
      "values": Array [
        true,
        true,
        "form-check-label ",
      ],
    },
    undefined,
  ],
}
`;

exports[`Storyshots Forms/CheckBox Group 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "
         <script>",
    "</script>
         ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
        <fieldset>
            <legend>Group Label</legend>
            ",
        "
        </fieldset>
        ",
      ],
      "values": Array [
        Array [
          Object {
            "_$litType$": 1,
            "strings": Array [
              "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
              "  ?disabled=",
              " id=\\"customCheck",
              "\\" />
                <label class=",
              " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>",
            ],
            "values": Array [
              false,
              false,
              1,
              "form-check-label ",
            ],
          },
          Object {
            "_$litType$": 1,
            "strings": Array [
              "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
              "  ?disabled=",
              " id=\\"customCheck",
              "\\" />
                <label class=",
              " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>",
            ],
            "values": Array [
              false,
              false,
              2,
              "form-check-label ",
            ],
          },
          Object {
            "_$litType$": 1,
            "strings": Array [
              "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
              "  ?disabled=",
              " id=\\"customCheck",
              "\\" />
                <label class=",
              " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>",
            ],
            "values": Array [
              false,
              false,
              3,
              "form-check-label ",
            ],
          },
          Object {
            "_$litType$": 1,
            "strings": Array [
              "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
              "  ?disabled=",
              " id=\\"customCheck",
              "\\" />
                <label class=",
              " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>",
            ],
            "values": Array [
              false,
              false,
              4,
              "form-check-label ",
            ],
          },
          Object {
            "_$litType$": 1,
            "strings": Array [
              "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
              "  ?disabled=",
              " id=\\"customCheck",
              "\\" />
                <label class=",
              " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>",
            ],
            "values": Array [
              false,
              false,
              5,
              "form-check-label ",
            ],
          },
        ],
      ],
    },
    undefined,
  ],
}
`;

exports[`Storyshots Forms/CheckBox Indeterminate 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "
         <script>",
    "</script>
         ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
        "  ?disabled=",
        " id=\\"customCheck111-33\\" />
                <label class=",
        " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>
        ",
      ],
      "values": Array [
        false,
        false,
        "form-check-label ",
      ],
    },
    "
      document.getElementById('customCheck111-33').indeterminate = true;
    ",
  ],
}
`;

exports[`Storyshots Forms/CheckBox Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "
         <script>",
    "</script>
         ",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
            <div class=\\"form-check\\">
                <input type=\\"checkbox\\" class=\\"form-check-input\\" ?checked=",
        "  ?disabled=",
        " id=\\"customCheck111-33\\" />
                <label class=",
        " for=\\"customCheck111-33\\">Check this custom checkbox</label>
            </div>
        ",
      ],
      "values": Array [
        false,
        false,
        "form-check-label ",
      ],
    },
    undefined,
  ],
}
`;
